### PR TITLE
[Feature]: New Theme Command: `shopify theme rename`

### PR DESCRIPTION
**Pull request by jamesmengo**
### WHY are these changes introduced?
- https://github.com/Shopify/cli/issues/1454

### WHAT is this pull request doing?
- Introduces a new command, `shopify theme rename`, which provides a direct command to rename a remote shopify theme
- This simplifies the current renaming workflow of:
  1) Pushing a theme with new name
  2) Deleting the old theme
  
Flags:
- `d` / `--development` -> specify whether or not to target the development theme. Note that this will override `t`
- `t` / `--theme` -> specify theme ID or name
- `n` / `-name` -> required

### How to test your changes?
- Rename theme: `pnpm shopify theme rename -n <new them name>`
- Rename development theme:`pnpm shopify theme rename -d -n <new theme name>`
- Rename a specific theme: `pnpm shopify theme rename -t <theme name or ID> -n <new theme name>`

## 🎩 
Nonexistent Theme ID:
image

Nonexistent Theme Name:
image

Specify Theme ID
image

Specify Theme Name
image

Select a Theme
image

Rename Development Theme
image

### Measuring impact

How do we know this change was effective? Please choose one:

- [ ] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [x] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes
- [x] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: packages/theme/src/cli/services/rename.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  const theme = await findOrSelectTheme(adminSession, {
+    header: 'Select a theme to open',
+    filter: {
+      theme: options.theme,
+      development: options.development,
+    },
+  })
+  await updateTheme(theme.id, {name: options.newName}, adminSession)
+  renderSuccess({
+    body: ['The theme', ...themeComponent(theme), 'was renamed to', options.newName],
```
Comment (isaacroldan) [inline packages/theme/src/cli/services/rename.ts:23]: I think it be easier to read if the old and new names are displayed in quotes
`The theme old name was renamed to new name`
vs
`The theme "old name" was renamed to "new name"`

What do you think?

**Review by isaacroldan — APPROVED** [review]

File: .changeset/late-beers-accept.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/theme': minor
+---
+
+Introduce `shopify theme rename` command
```
Comment (charlespwd) [inline .changeset/late-beers-accept.md:5]:
```suggestion
Introduce the `shopify theme rename` command
```

File: packages/theme/src/cli/commands/theme/rename.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    const store = ensureThemeStore(flags)
+    const adminSession = await ensureAuthenticatedThemes(store, password)
+
+    const renameOptions: RenameOptions = {
+      development,
+      newName: name,
+      theme,
+    }
+
+    await renameTheme(adminSession, renameOptions)
```
Comment (charlespwd) [inline packages/theme/src/cli/commands/theme/rename.ts:47]: What's the behaviour when this fails? What if a theme doesn't exist or something? Does this throw an error? Is there a mechanism to make sure that the exit code of the command is 1 when there is an error and 0 when the command was successful?

**Review by charlespwd — COMMENTED** [review]
Good stuff! Clarifications pls :)

File: packages/theme/src/cli/commands/theme/rename.ts (hunk 3 above)
Comment (jamesmengo) [reply packages/theme/src/cli/commands/theme/rename.ts:47]: image

File: packages/theme/src/cli/commands/theme/rename.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import {ensureThemeStore} from '../../utilities/theme-store.js'
+import {themeFlags} from '../../flags.js'
+import {Flags} from '@oclif/core'
+import {globalFlags} from '@shopify/cli-kit/node/cli'
+import {ensureAuthenticatedThemes} from '@shopify/cli-kit/node/session'
+
+export default class Rename extends ThemeCommand {
+  static description = 'Renames an existing theme.'
+
+  static flags = {
```
Comment (karreiro) [inline packages/theme/src/cli/commands/theme/rename.ts:12]: May we support the `--live` flag as well? In the other commands, we generally support `-d`/`-t`/`-l` for theme selection :)

**Pull request merged**